src/odemis/driver/xt_client.py

```diff
@@ -349,20 +468,579 @@ def set_autofocusing(self, channel, state):
+            setter=self._setRotation)
+
+        scanning_size_info = self.parent.scanning_size_info()
+        fov = self.parent.get_scanning_size()[0]
+        self.horizontalFoV = model.FloatContinuous(
+            fov,
+            unit=scanning_size_info["unit"],
+            range=scanning_size_info["range"]["x"],
+            setter=self._setHorizontalFoV)
+
+        mag = self._hfw_nomag / fov
+        mag_range_max = self._hfw_nomag / scanning_size_info["range"]["x"][0]
+        mag_range_min = self._hfw_nomag / scanning_size_info["range"]["x"][1]
+        self.magnification = model.FloatContinuous(mag, unit="",
+                                                   range=(mag_range_min, mag_range_max),
+                                                   readonly=True)
+        # To provide some rough idea of the step size when changing focus
+        # Depends on the pixelSize, so will be updated whenever the HFW changes
+        self.depthOfField = model.FloatContinuous(1e-6, range=(0, 1e3),
+                                                  unit="m", readonly=True)
+        self._updateDepthOfField()
+
+        # Refresh regularly the values, from the hardware, starting from now
+        self._updateSettings()
+        self._va_poll = util.RepeatingTimer(5, self._updateSettings, "Settings polling")
+        self._va_poll.start()
+
+    def _updateSettings(self):
+        """
+        Read all the current settings from the SEM and reflects them on the VAs
+        """
+        logging.debug("Updating SEM settings")
+        try:
+            dwell_time = self.parent.get_dwell_time()
+            if dwell_time != self.dwellTime.value:
+                self.dwellTime._value = dwell_time
+                self.dwellTime.notify(dwell_time)
+            voltage = self.parent.get_ht_voltage()
+            if voltage != self.accelVoltage.value:
+                self.accelVoltage._value = voltage
+                self.accelVoltage.notify(voltage)
+            blanked = self.parent.beam_is_blanked()
+            if blanked != self.blanker.value:
+                self.blanker._value = blanked
+                self.blanker.notify(blanked)
+            spot_size = self.parent.get_ebeam_spotsize()
+            if spot_size != self.spotSize.value:
+                self.spotSize._value = spot_size
+                self.spotSize.notify(spot_size)
+            beam_shift = self.parent.get_beam_shift()
+            if beam_shift != self.beamShift.value:
+                self.beamShift._value = beam_shift
+                self.beamShift.notify(beam_shift)
+            rotation = self.parent.get_rotation()
+            if rotation != self.rotation.value:
+                self.rotation._value = rotation
+                self.rotation.notify(rotation)
+            fov = self.parent.get_scanning_size()[0]
+            if fov != self.horizontalFoV.value:
+                self.horizontalFoV._value = fov
+                self.horizontalFoV.notify(fov)
```

Comment:
It'd slightly better to only notify after the magnification value has been set. This way, if a subscriber reads the magnification value, it'll already be correct.